cohortextractor/cohortextractor.py

```diff
@@ -772,6 +774,21 @@ def main():
         except DummyDataValidationError as e:
             print(f"Dummy data error: {e}")
             sys.exit(1)
+        except database_exceptions() as e:
```

Comment:
This is interesting, I've never seen except clauses dynamically generated before, but now I think of it, it seems plausible.

I did find this surprising, though. Would this work if we do the try/catch at the module level, and define this list at import time, rather than re-checking every time?

Reply:
So it's like this because otherwise a test fails if you import pymssql at the module level.  I think it'll still complain if the try/catch is at the modue level, because it'll have pymssql in sys.modules

Reply:
Could we cache the result of the function at module level, but leave the import inside the function scope?

e.g.
```
DATABASE_EXCEPTIONS = database_exceptions()
```
then
```
        except DATABASE_EXCEPTIONS as e:
```